presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestDistributedQueries.java

```diff
@@ -765,6 +767,21 @@ public void testQueryLoggingCount()
         });
     }
 
+    private <T> T waitUtilStable(Supplier<T> fun, Duration timeout)
+    {
+        Optional<T> lastValue = Optional.empty();
+        long start = System.nanoTime();
+        while (!currentThread().isInterrupted() && nanosSince(start).compareTo(timeout) < 0) {
+            T currentValue = fun.get();
+            if (lastValue.isPresent() && currentValue.equals(lastValue.get())) {
```

Comment:
You could
```
lastValue = fun.get()
```
and then sleep before getting next value. This allow you to live without `Optional`